### DISCUSSION TITLE
React testing library - how to test client side using the query client

### DISCUSSION THREAD

**Discussion by shannonhochkins** [trpc/trpc#3429] [Q&A]
Whenever i mount a component using the query client, i keep getting `TRPCClientError: No fetch implementation found` which is clearly becuase i need to mock a fetch implementation for jest, however the url structure sent to trpc especially in batch mode can be quite complicated.

I was wondering if anyone has any solutions on how to mock requests for trpc in a nice way?

Realistically, i cannot find a single example online of how to test the client side react application with trpc and I would absolutely love some guidance!

I've seen others say "mock the db not the endpoints" but even localised, it's still going to have to make a request to the backend of trpc and then makes the tests pretty hard to manage as opposed to just mocking the expected response right in the test

I just cannot for the life of me figure out how to do so!

**Comment by erikhansenmujica** [comment]
Helloo, did you have any advance on this? I am having the same problem :(

**Comment by mwimbledon** [comment]
Same problem here, If anyone has a solution for this I'd be interested to hear it
